c2corg_api/search/notify_sync.py

```diff
@@ -0,0 +1,40 @@
+import transaction
+import logging
+from kombu.pools import producers
+
+
+log = logging.getLogger(__name__)
+
+
+def notify_es_syncer(queue_config):
+    """Notify the syncer script that a document has changed by pushing a
+    friendly message into the Redis queue.
+    """
+    def push_notification():
+        with producers[queue_config.connection].acquire(
+                block=True, timeout=3) as producer:
```

Comment:
> When can we get a timeout?
> 
>     no more connections in pool (all 200 connections active; do we need so many of them?);

Good point, I think we don't need that many.

>     dead/unresponsive syncher

At this point you know nothing about the syncer. You only talk to Redis.

>  (will it restart automatically? I think there is a sync on start)

Right now it does not restart automatically in case it died somehow. This is a point I have to check with Mark.

> Are there other cases?

When Redis is down.